hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -847,6 +868,123 @@ class Emit[C](
         assert(unified)
         impl.applySeededI(seed, cb, EmitRegion(mb, region), pt, codeArgs: _*)
 
+      case x@ShuffleWith(
+        keyFields,
+        rowType,
+        rowEType,
+        keyEType,
+        name,
+        writerIR,
+        readersIR
+      ) =>
+        val shuffleType = x.shuffleType
+        val shufflePType = x.shufflePType
+
+        val (socket, in, out, log) = ShuffleClient.openConnection(cb, shuffleType)
```

Comment:
I'm having trouble commenting in our previous threads, so I'll ask this again here: it seems like you've inlined (part of) the shuffle client constructor and the implementation of ShuffleClient.start and ShuffleClient.stop.  I'm wondering why this code isn't just:

```
  val shuffleClient = CodeShuffleClient.create(cb, getPType(shuffleType))
  shuffleClient.start(cb)
  ...
  ...withCleanup {
    shuffleClient.stop(cb)
  }
```

Reply:
ok